src/pages/confirmed.js

```diff
@@ -0,0 +1,115 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import Header from '../components/Header';
+import Footer from '../components/Footer';
+import { fetchOrder } from '../lib/opencollective';
+import ConfirmationFAQ from '../../ConfirmationFAQ.md';
+
+export default class Confirmed extends React.Component {
+  static getInitialProps({ query }) {
+    return {
+      next: query.next || '/',
+      orderId: query.orderId,
+    };
+  }
+
+  static propTypes = {
+    loggedInUser: PropTypes.object,
+    orderId: PropTypes.string,
+  };
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      dispatchedOrders: [],
+    };
+  }
+
+  async componentDidMount() {
+    if (!this.props.orderId) {
+      return;
+    }
+    const order = await fetchOrder(parseInt(this.props.orderId));
```

Comment:
Instead of fetching the order here, we should execute the `dispatchOrder` mutation from there and collect the result.